addons/mail/models/mail_thread.py

```diff
@@ -4519,53 +4519,46 @@ def _message_update_content(self, message, body, attachment_ids=None, partner_id
         empty_messages = message.sudo()._filter_empty()
         empty_messages._cleanup_side_records()
         empty_messages.write({'pinned_at': None})
-        res = {
-            "attachment_ids": Store.many(message.attachment_ids.sorted("id")),
-            "body": message.body,
-            "pinned_at": message.pinned_at,
-            "recipients": Store.many(message.partner_ids, fields=["name", "write_date"]),
-            "write_date": message.write_date,
-        }
+        res = [
+            Store.Many("attachment_ids", sort="id"),
+            "body",
+            Store.Many("partner_ids", ["name", "write_date"], rename="recipients"),
+            "pinned_at",
+            "write_date",
+        ]
         if body is not None:
             # sudo: mail.message.translation - discarding translations of message after editing it
             self.env["mail.message.translation"].sudo().search([("message_id", "=", message.id)]).unlink()
-            res["translationValue"] = False
+            res.append(Store.Attr("translationValue", False))
```

Comment:
Much simpler :brain:  !

Reply:
Oversight from the bigger changes, this one can be changed back to a key/value pair with a little tweak: https://github.com/odoo/odoo/pull/190230